sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/storage/PreferenceTokenStorage.java

```diff
@@ -0,0 +1,83 @@
+package com.microsoft.appcenter.identity.storage;
+
+import android.content.Context;
+import android.support.annotation.VisibleForTesting;
+
+import com.microsoft.appcenter.utils.context.AuthTokenContext;
+import com.microsoft.appcenter.utils.crypto.CryptoUtils;
+import com.microsoft.appcenter.utils.storage.SharedPreferencesManager;
+
+/**
+ * Storage for tokens that uses {@link SharedPreferencesManager}. Handles saving and encryption.
+ */
+public class PreferenceTokenStorage implements AuthTokenStorage {
```

Comment:
If this class is also storing account ID and not just auth token I think that we could do some renaming.  Perhaps `PreferenceAccountStorage` or `PreferenceAuthStorage`?  I'm not in love with either of those, does anyone else have thoughts/ideas?  If we change it, the methods (`getToken`, `saveToken`, etc.) should be changed to align.

Reply:
Or maybe this is an OK convention... I don't know.  Does anyone have another opinion?

Reply:
@jwallra i think storing last account id doesn't change the main "token storage" purpose. I mean this class still focuses mainly on token and uses account id just as a piece of a helper info to tell when the user has changed. It also does not provide any publuc methods for retrieving/removing account id etc.

Reply:
userId is a claim within the token.